File: 스프링_리액티브_프로그래밍/part3/chapter18/src/main/java/com/itvillage/book/v5/BookRepository.java
package com.itvillage.book.v5;

import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Mono;

@Repository("bookRepositoryV5")
public interface BookRepository extends ReactiveCrudRepository<Book, Long> {
    Mono<Book> findByIsbn(String isbn);
}

File: cloud_native_spring/demo/catalog-service/src/main/kotlin/com/yscorp/catalog/domain/BookRepository.kt
package com.yscorp.catalog.domain

import org.springframework.data.jdbc.repository.query.Modifying
import org.springframework.data.jdbc.repository.query.Query
import org.springframework.data.repository.CrudRepository
import org.springframework.transaction.annotation.Transactional
import java.util.*

interface BookRepository : CrudRepository<Book, Long> {

    fun findByIsbn(isbn: String): Optional<Book>

    fun existsByIsbn(isbn: String): Boolean

    @Modifying
    @Transactional
    @Query("delete from Book where isbn = :isbn")
    fun deleteByIsbn(isbn: String)
}